src/nlp_utils.jl

```diff
@@ -23,3 +23,37 @@ macro rangecheck(lo, hi, vars...)
   end
   Expr(:block, exprs...)
 end
+
+"""
+    coo_prod!(nrows, nnz, rows, cols, vals, v, Av)
+
+Compute the matrix product `Av` where `A` is given by `(rows, cols, vals)` with `nrows` rows.
+The dimensions of `rows`, `cols` and `vals` should be greater of equal to `nnz`.
+"""
+function coo_prod!(nrows :: Integer, nnz :: Integer, rows :: AbstractVector{<: Integer}, cols :: AbstractVector{<: Integer}, vals :: AbstractVector, v :: AbstractVector, Av :: AbstractVector)
+  Av[1:nrows] .= zero(eltype(v))
+  @inbounds for k = 1:nnz
```

Comment:
If we deduce nnzj, we'll never risk going out of bounds.

Reply:
Ok, I can deduce `nnzj`. In this case, makes sense assume `Av` is the correct size as well, right? I can simply do `Av .= 0`, or `fill!`.

Reply:
Yes I guess so, but let's document that clearly.